app/src/main/java/com/example/drawingactivity/ColorUtils.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+object ColorUtils {
+
+    private val HEXADECIMAL_PATTERN: Pattern = Pattern.compile("\\p{XDigit}+")
+
+    /**
+     * A data class that contains color values.
+     * Contains: Hue, Saturation, Value, Red, Green, Blue, and Hex Value.
+     */
+    data class ColorState(
+        var hue: Int,
+        var saturation: Float,
+        var value: Float,
+        var red: Int,
+        var green: Int,
+        var blue: Int,
+        var hexCode: String
+    )
+
+    /**
+     * Creates a color state out of the supplied hex code.
+     * Returns a color state containing all of the values
+     */
+    fun createColorStateFromHexCode(hexCode: String): ColorState {
+        val color = Color.parseColor(hexCode)
+        val rgb = IntArray(3)
+        rgb[0] = Color.red(color)
+        rgb[1] = Color.green(color)
+        rgb[2] = Color.blue(color)
+        return calculateColorWithRGB(rgb)
+    }
+
+    /**
+     * Calculates the RGB of a color using a HSV float array.
+     * Returns a ColorState containing all values.
+     */
+    fun calculateColorWithHSV(hsv: FloatArray): ColorState {
+        val currentColor: Int = Color.HSVToColor(hsv)
+        val rgbArray = IntArray(3)
+        rgbArray[0] = Color.red(currentColor)
+        rgbArray[1] = Color.green(currentColor)
+        rgbArray[2] = Color.blue(currentColor)
+        val hex = java.lang.String.format("#%02x%02x%02x", rgbArray[0], rgbArray[1], rgbArray[2])
+        return ColorState(
+            hue = hsv[0].toInt(),
+            saturation = hsv[1],
+            value = hsv[2],
+            red = rgbArray[0],
+            green = rgbArray[1],
+            blue = rgbArray[2],
+            hexCode = hex
+        )
+    }
+
+    /**
+     * Calculates the HSV of a color using a RGB Int array.
+     * Returns a ColorState containing all values.
+     */
+    fun calculateColorWithRGB(rgb: IntArray): ColorState {
+        val hsv = FloatArray(3)
+        val currentColor = Color.rgb(rgb[0], rgb[1], rgb[2])
+        Color.colorToHSV(currentColor, hsv)
+        val hex = java.lang.String.format("#%02x%02x%02x", rgb[0], rgb[1], rgb[2])
+        return ColorState(
+            hue = hsv[0].toInt(),
+            saturation = hsv[1],
+            value = hsv[2],
+            red = rgb[0],
+            green = rgb[1],
+            blue = rgb[2],
+            hexCode = hex
+        )
+    }
+
+    /**
+     * Function that returns a boolean based on if the provided string contains only numbers.
+     */
+    fun isIntegerInput(text: String): Boolean {
+        return text.toIntOrNull() != null
+    }
+
+    /**
+     * Function that returns a boolean based on if the string matches a hexadecimal pattern.
+     */
+    fun isHexadecimal(input: String): Boolean {
```

Comment:
if this is similar to isIntegerInput may want to consider previous comment and apply it to this, if it's not (which it looks slightly different) then ignore this comment.

Reply:
this also converts the input to a hex string format, so it does make sense to me to have this as a function. Removed it anyways since it no longer is used for anything